app/services/order_cycle_distributed_products.rb

```diff
@@ -13,6 +13,21 @@ def products_relation
     Spree::Product.where(id: stocked_products).group("spree_products.id")
   end
 
+  # Joins on the first product variant to allow us to filter product by taxon. This is so
+  # enterprise can display product sorted by category in a custom order on their shopfront.
+  #
+  # Caveat, the category sorting won't work properly if there are multiple variant with different
+  # category for a given product.
+  #
+  def products_taxons_relation
+    Spree::Product.where(id: stocked_products).
+      joins("LEFT JOIN (SELECT DISTINCT ON(product_id) id, product_id, primary_taxon_id
+                        FROM spree_variants) first_variant
+                        ON spree_products.id = first_variant.product_id").
```

Comment:
I was wondering how we select only the first variant. After a while I realised that `DISTINCT ON(product_id)` ensures that we select only one variant per product but I guess that it could be a random one.

Reply:
We could add an `ORDER BY created_at` on the sub query and ignore deleted ones so we always pick up the first created one ?

Reply:
I don't think it matters which one we select as long as we ignore deleted ones.